### ISSUE TITLE
using provider metadata (tags, ...)

### ISSUE THREAD

**Issue by moolen** [external-secrets/external-secrets#305]
Multiple users reported a use-case to sync tags or arbitrary metadata from the secret provider. Yaml spec is defined in comment below.

Implementation status:

- [x] Azure KV
- [ ] AWS Secrets Manager
- [ ] AWS Parameter Store
- [ ] GCP Secrets Manager
- [ ] Hashicorp Vault
- [x] Kubernetes

**Comment by moolen** [comment]
The CRD changes were implemented in #1062 and look like this:

```yaml
apiVersion: external-secrets.io/v1beta1
kind: ExternalSecret
metadata:
  name: example-external-secret
spec:
  # [...]
  data:
  # fetch all the tags in JSON format
  - secretKey: dev-secret-test
    remoteRef:
      key: dev-secret-test
      metadataPolicy: Fetch

  # fetch a specific tag which name must be in property
  - secretKey: dev-secret-test
    remoteRef:
      key: dev-secret-test
      metadataPolicy: Fetch
      property: tagname

  dataFrom:
  # similar like above, but returns an map[string]string.
  - extract: 
      key: test
      metadataPolicy: Fetch
```
I'll track the implementation in the first comment :point_up_2:.

**Comment by moolen** [comment]
duplicate of #1872
